### PR TITLE
design(mu_cosine): Haiku-expectation training, overlap-aware partition, and sampling mechanics (§9–§11)

### PR DESCRIPTION
**Pull request by s243a**
Three design sections that settle how the inferred-operator superposition is trained, evaluated, and
  reasoned about — the conclusion of the anchored-basis design arc. **Docs only** (the implementation +
  its A/B/ablation evidence live on `claude/mu-anchored-basis-wiring`).

  ### §9 — Haiku-expectation training + eval (the settled scheme)
  - **Train on the expectation, not the full per-operator distribution** (the deliverable is the read-out μ;
    the elaborations that leaned on explicit `P(op)` didn't beat the simple version — see the wiring
    branch's ablation). Reuses the infer-blend.
  - **Stochastic 70/30 target:** per row/step, prob `c≈0.7` → the **label**'s μ; prob `(1-c)` → **Haiku**'s
    `E[μ]`. Both directions for asymmetric ops.
  - **Eval against Haiku *alone*** (independent → no circularity) over the **inferred tail** — the population
    SYM/disc never measured.
  - **`none` anchor** (μ≈0) for true negatives, distinct from the learnable atoms (novel positives).
  - Haiku scored **once + cached**, scoped to the inferred tail; one pass serves both the training target
    and the eval reference.

  ### §10 — Overlap-aware partition (proposed)
  - **Conceptual closure:** relations correlate (not truly disjoint), but the softmax acts on the *evidence
    space* and imposes a **learned effective partition** — correlation lives in the soft boundaries / overlap
    atoms, sharpness = learnable effective-disjointness, so `E[μ]` stays a bounded convex combination
    regardless of correlation.
  - **Proposed design:** the disjoint-event partition *is* the anchored basis — singleton **anchors** +
    overlap/novel **atoms** + **`none`** cell; softmax = the partition (Σ=1); `E[μ]=ΣP(cell)·μ_cell`. Mutual
    exclusivity is the special case; the multi-label generalisation **promotes an overlap atom** when a region
    persistently splits weight between two anchors with a distinct μ (the §8b grow rule for overlaps).

  ### §11 — Training mechanics (corrects an earlier error)
  - μ is **non-linear** in the operator weights (the op token passes through the non-linear transformer), so
    `μ(mean) ≠ E[μ]` (Jensen) — feeding one averaged input is a biased shortcut. **We sample.**
  - **For training, coverage beats exact correlations:** random Haiku-informed superpositions cover the space;
    knowing the correlations just makes samples better-placed, not required — training isn't gated on them.
  - **For inference `E[μ]`,** the correlations *do* matter: Monte-Carlo average over samples drawn from
    `P(op|pair)` (the §10 partition).

  ### Status
  Design is complete and self-consistent: §1b (attention/probability) → §7 (regularizer) → §8/8b/8c (anchored
  basis) → §9 → §10 → §11. The architecture is **validated** (anchored-μ beats baseline, twice — wiring
  branch); the one remaining runnable step is a **single budget-gated Haiku pass** over the inferred tail to
  turn §9–§11 into an experiment.

### CODE REVIEW

File: prototypes/mu_cosine/DESIGN_inferred_operator_superposition.md [hunk 1]
```diff
@@ -334,3 +334,108 @@ Fusion: `q = q_proj([μ_vec ++ provenance ++ e5_raw])` → attend over the ancho
...
+
+**Train on the EXPECTATION, not the full distribution.** The deliverable is the read-out μ, so train the
+*blended* μ toward an expectation target, reusing the infer-blend (op_weights override + MSE). We do NOT
+supervise `P(op)` per-operator (the elaborations that leaned on it didn't pay off). Trade accepted: we lose
+the direct `P(op)_model` vs `P(op)_ref` eval and the explicit decomposition — but optimise the thing we use.
+
+**Stochastic 70/30 target (per row, per step):**
+- with prob = the row's label-confidence `c` (≈0.7): target the **label's** μ (calibrated, both directions);
+- with prob `(1-c)`: target **Haiku's expectation** `E[μ] = Σ P_Haiku(op)·μ_op` (both directions).
+This is the Monte-Carlo realisation of `c·label + (1-c)·Haiku` applied to the μ expectation. Keep
```
Comment (s243a) [inline prototypes/mu_cosine/DESIGN_inferred_operator_superposition.md:351]: **[blocking-for-design] Keep the Haiku expectation at relation/cell level, not operator level**

This writes `E[μ] = Σ P_Haiku(op)·μ_op`, which appears to re-collapse the many-to-one operator mapping that §5b/#3359 deliberately kept at relation level. In particular, `SYM` can cover relations with different μ semantics, so an operator-level expectation loses the distinction. Please write this over relations/cells, e.g. `Σ_cell P_Haiku(cell)·μ_cell`, or spell out the relation→operator marginalization and why it is safe for SYM/overlap cells.

File: prototypes/mu_cosine/DESIGN_inferred_operator_superposition.md [hunk 2]
```diff
@@ -334,3 +334,108 @@ Fusion: `q = q_proj([μ_vec ++ provenance ++ e5_raw])` → attend over the ancho
...
+transformer** (attention + MLP). Because that path is non-linear, **the output of the *averaged* input is NOT
+the average of the outputs** — feeding the mean superposition gives `μ(mean)`, which is *not* the expectation
+`E[μ] = Σ P·μ`. (This is Jensen's inequality, and it's the §1 caveat: the blended-mean is a *biased shortcut*,
+not the true expectation.) Each *sampled* superposition is its own non-linear forward; they don't collapse to
+the mean. So we **sample**.
+
+**Training — the §9 70/30, made precise:**
+- **70% of steps:** train on the **label** (its operator + μ).
+- **30% of steps:** train on a **random superposition informed (at least in part) by Haiku** — draw a
+  superposition shaped by Haiku's distribution, feed it, backprop on its target.
```
Comment (s243a) [inline prototypes/mu_cosine/DESIGN_inferred_operator_superposition.md:436]: **[blocking-for-design] The 30% Haiku branch is still under-specified**

§9 says the Haiku leg targets an aggregate expectation, while §11 says to feed a random Haiku-informed superposition and backprop on “its target.” Those are different realizations through a nonlinear model. Please make one scheme canonical and define the per-sample target explicitly: is it the aggregate Haiku `E[μ]`, the sampled operator/cell's calibrated μ, or a convex target `Σ_i w_i μ_i` tied to the sampled superposition? Without that contract, the wiring branch can implement multiple incompatible algorithms while claiming to follow the same spec.

File: prototypes/mu_cosine/DESIGN_inferred_operator_superposition.md [hunk 3]
```diff
@@ -334,3 +334,108 @@ Fusion: `q = q_proj([μ_vec ++ provenance ++ e5_raw])` → attend over the ancho
...
+- with prob `(1-c)`: target **Haiku's expectation** `E[μ] = Σ P_Haiku(op)·μ_op` (both directions).
+This is the Monte-Carlo realisation of `c·label + (1-c)·Haiku` applied to the μ expectation. Keep
+`op_weights` and the target on the **same** mixture (mismatching them distorts the per-op μ's).
+
+**Asymmetric operators → both permutations:** train `E[μ(node|root)]` and `E[μ(root|node)]` separately
+(REL_SPEC already stores fwd/rev).
+
+**Eval against Haiku ALONE (not the training mixture).** Using the label-biased mixture as the eval reference
+is circular (the model trained on the label). Independent reference = `corr(E[μ]_model, E[μ]_Haiku)` over the
+**inferred** held-out rows — the population the anchored basis exists for, which SYM/disc never measured.
```
Comment (s243a) [inline prototypes/mu_cosine/DESIGN_inferred_operator_superposition.md:363]: **[high] “Haiku alone” is not fully independent if Haiku is also a training target**

Evaluating against Haiku instead of the label mixture avoids one circularity, but it does not make the reference independent if Haiku supplies the 30% training target on the same inferred-tail rows. The later “same pass serves both the training target and eval reference” sentence makes this sound like measuring fit to the cached teacher, not held-out generalization. Please define the cache/split protocol and weaken the claim to something like “teacher-consistency on held-out Haiku-scored rows,” unless the eval rows are excluded from all Haiku-target training.

File: prototypes/mu_cosine/DESIGN_inferred_operator_superposition.md [hunk 4]
```diff
@@ -334,3 +334,108 @@ Fusion: `q = q_proj([μ_vec ++ provenance ++ e5_raw])` → attend over the ancho
...
+After the anchored-basis A/B (architecture validated; the explicit per-operator distribution and the
+text-in-query did NOT beat the simple version — `REPORT_anchored_basis_ab.md`), the operating scheme:
+
+**Train on the EXPECTATION, not the full distribution.** The deliverable is the read-out μ, so train the
+*blended* μ toward an expectation target, reusing the infer-blend (op_weights override + MSE). We do NOT
+supervise `P(op)` per-operator (the elaborations that leaned on it didn't pay off). Trade accepted: we lose
+the direct `P(op)_model` vs `P(op)_ref` eval and the explicit decomposition — but optimise the thing we use.
+
+**Stochastic 70/30 target (per row, per step):**
+- with prob = the row's label-confidence `c` (≈0.7): target the **label's** μ (calibrated, both directions);
```
Comment (s243a) [inline prototypes/mu_cosine/DESIGN_inferred_operator_superposition.md:349]: **[medium] Restate isolated-RNG discipline for the new stochastic switch**

The per-row/per-step 70/30 target choice and the §11 superposition sampler introduce the same class of stochastic intervention that caused the earlier infer-switch A/B confound before RNG isolation. Please state that these draws use isolated, seedable RNG streams, separate from data order, model initialization, dropout/masking, and any eval sampling.

File: prototypes/mu_cosine/DESIGN_inferred_operator_superposition.md [hunk 5]
```diff
@@ -334,3 +334,108 @@ Fusion: `q = q_proj([μ_vec ++ provenance ++ e5_raw])` → attend over the ancho
...
+  as fuzzy boundaries / dedicated overlap cells — never a contradiction.
+- The softmax **sharpness = how disjoint to treat them**, and it is *learnable* — correlated relations get
+  softer boundaries, separable ones sharp boundaries; tuned per region.
+- So `E[μ] = Σ w·μ` is a **smooth mixture mean over the learned partition**, bounded in [0,1] *regardless* of
+  the true correlation. The categorical is valid because the softmax *manufactures* the effective
+  disjointness in evidence-space.
+
+### Proposed design: the disjoint-event partition
+Generalise the operator distribution from "one relation per pair" to a **disjoint partition of the
+relation-combination space** (the joint over relation *subsets*), realised directly by the anchored basis:
```
Comment (s243a) [inline prototypes/mu_cosine/DESIGN_inferred_operator_superposition.md:402]: **[medium] Define the overlap-aware grouping/partition key operationally**

This section gives a useful conceptual interpretation, but “overlap-aware partition” is not falsifiable until the implementation defines the grouping/overlap key: source program, benchmark, structural hash, pair identity, semantic-near-duplicate threshold, or something else. Please add the key and the intended train/eval split rule so downstream evidence can distinguish leakage-controlled generalization from within-group interpolation.

File: prototypes/mu_cosine/DESIGN_inferred_operator_superposition.md [hunk 6]
```diff
@@ -334,3 +334,108 @@ Fusion: `q = q_proj([μ_vec ++ provenance ++ e5_raw])` → attend over the ancho
...
+
+**Coverage beats exact correlations — for *training*.** Training only needs good **coverage** of the
+superposition space so the model learns the (non-linear) function across it. It does **not** need the exact
+joint / correlations. Random Haiku-informed superpositions already give coverage; **knowing the correlations
+just makes each sample more informative (better-placed), not required.** The §10 partition / exact
+correlations matter most for an accurate *inference* `E[μ]`, far less for training coverage — so we don't
+gate training on having them.
+
+**Inference `E[μ]`** (where correlations *do* matter): the true expectation is the **Monte-Carlo average over
+sampled superpositions**, not a single mean-fed pass (that pass is fast but Jensen-biased). For the
```
Comment (s243a) [inline prototypes/mu_cosine/DESIGN_inferred_operator_superposition.md:446]: **[medium] Add an MC sample-count/error-bar contract for inference `E[μ]`**

If inference `E[μ]` is a Monte-Carlo average over sampled superpositions, the design should name the sample count `N`, the sampling distribution, the RNG/seed policy, and the expected error reporting. Otherwise a future implementation could use `N=1` and still claim to estimate an expectation, even though variance/error bars would be uncontrolled.

**Review by s243a — COMMENTED** [review]
Council review summary: **needs discussion**. Opus 4.8, GPT-5.5, Gemini, and the deep-research pass all agree the §9→§11 arc is coherent and useful, but this should not become the reference spec until two design contracts are made unambiguous: (1) what exactly happens on the 30% Haiku branch, and (2) what independence/evaluation claim is valid when Haiku supplies both training signal and eval reference.

Key asks:
- Rewrite Haiku expectations over relations/cells rather than operators, or explicitly define the relation→operator marginalization so SYM/overlap cases do not collapse distinct relation μ targets.
- Make §9 and §11 agree on whether the Haiku leg trains against an aggregate target, a sampled superposition, or a per-sample convex target.
- Weaken the “Haiku alone = independent/no circularity” claim and define the held-out/cache protocol.
- Carry forward #3356-style isolated RNG for the new 70/30 and superposition sampling draws.
- Specify the overlap-aware grouping key and MC sample-count/error-bar protocol before the wiring branch reports numbers against this design.

External context reinforces these caveats: LLM-generated labels carry correlated teacher errors, LLM-as-judge can exhibit preference leakage, stochastic-label unbiasedness depends on the sampling mechanism, group-aware splits are the standard leakage control for correlated examples, and Monte Carlo expectation estimates need sample-count/error-bar discipline.
